banyand/internal/storage/index.go

```diff
@@ -224,3 +228,115 @@ func appendSeriesList(dest, src pbv1.SeriesList, filter posting.List) pbv1.Serie
 func (s *seriesIndex) Close() error {
 	return s.store.Close()
 }
+
+type seriesIndexController[T TSTable, O any] struct {
+	ctx     context.Context
+	hot     *seriesIndex
+	standby *seriesIndex
+	timestamp.TimeRange
```

Comment:
What's the time range used for?

Reply:
it's used to mark the creation time of series index and closing time which reached the TTL.  When triggering the scheduler, it will compare      the end time of seriesIndexController and deadline. And if the end time is before deadline, it will switch standby and hot index, otherwise create a new series index

Reply:
You can remove the time range and leave only the end time.